core/tauri/src/updater/core.rs

```diff
@@ -833,54 +842,29 @@ fn copy_files_and_run<R: Read + Seek>(
   );
 
   let current_exe_args = env.args.clone();
+  let mut installer_args_common = config
+    .tauri
+    .updater
+    .windows
+    .installer_args
+    .iter()
+    .map(OsStr::new)
+    .collect::<Vec<_>>();
 
   for path in paths {
     let found_path = path?.path();
     // we support 2 type of files exe & msi for now
-    // If it's an `exe` we expect an installer not a runtime.
+    // If it's an `exe` we expect an NSIS installer.
     if found_path.extension() == Some(OsStr::new("exe")) {
-      // we need to wrap the installer path in quotes for Start-Process
-      let mut installer_path = std::ffi::OsString::new();
-      installer_path.push("\"");
-      installer_path.push(&found_path);
-      installer_path.push("\"");
-
-      let installer_args = [
-        config
-          .tauri
-          .updater
-          .windows
-          .install_mode
-          .nsis_args()
-          .iter()
-          .map(ToString::to_string)
-          .collect(),
-        vec!["/ARGS".to_string()],
-        current_exe_args,
-        config
-          .tauri
-          .updater
-          .windows
-          .installer_args
-          .iter()
-          .map(ToString::to_string)
-          .collect::<Vec<_>>(),
-      ]
-      .concat();
-
-      // Run the EXE
-      let mut cmd = Command::new(powershell_path);
-      cmd
-        .args(["-NoProfile", "-WindowStyle", "Hidden", "Start-Process"])
-        .arg(installer_path);
-      if !installer_args.is_empty() {
-        cmd.arg("-ArgumentList").arg(installer_args.join(", "));
-      }
-      cmd
-        .spawn()
-        .expect("Running NSIS installer from powershell has failed to start");
-
-      exit(0);
+      installer_args_common.extend(config
```

Comment:
we can't use extend here, order of arguments is important, install_mode args should be first.